File: 2924_Find_Champion_II.cpp
class Solution {
public:
    int findChampion(int n, vector<vector<int>>& edges) {
        vector<int> indegs(n, 0);
        for (auto e : edges)
            indegs[e[1]]++;

        int champ = -1;
        int cnt = 0;

        for (int i = 0; i < n; ++i)
        {
            if (indegs[i] == 0)
            {
                cnt++;
                champ = i;
            }
        }

        return cnt > 1 ? -1 : champ;
    }
};

File: 2924_Find_Champion_II.py
class Solution(object):
    def findChampion(self, n, edges):
        """
        :type n: int
        :type edges: List[List[int]]
        :rtype: int
        """
        indegs = [0] * n 
        for edge in edges:
            indegs[edge[1]] += 1

        champ = -1 
        cnt = 0

        for i in range(0, n):
            if indegs[i] == 0:
                champ = i
                cnt += 1

        return -1 if cnt > 1 else champ
